Week7/CurrentNews/app/src/test/java/com/florencenjeri/currentnews/viewmodel/NewsViewModelTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.florencenjeri.currentnews.viewmodel
+
+import androidx.lifecycle.MutableLiveData
+import com.florencenjeri.currentnews.App
+import com.florencenjeri.currentnews.database.NewsDatabase
+import com.florencenjeri.currentnews.database.NewsRepository
+import com.florencenjeri.currentnews.model.News
+import com.florencenjeri.currentnews.ui.viewmodel.NewsViewModel
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Spy
+import org.mockito.junit.MockitoJUnit
+import org.mockito.junit.MockitoRule
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest=Config.NONE)
+class NewsViewModelTest {
+    val newsDao by lazy { NewsDatabase.getDatabase(App.getAppContext()).newsDao() }
```

Comment:
I think you can remove a good part of this test, as you're not actually testing the DAO, repo, or anything else, but rather the interaction between the ViewModel and its dependencies. I'd also have added some `verify()` from Mockito, to check if those functions are being called or not.

Otherwise it's good.